### PR TITLE
Upload C++ and Python codecov reports artifacts

### PR DESCRIPTION
**Pull request by maliasadi**
**Context:**
The frontend and runtime Python and C++ code coverage reports can be generated from source using `make coverage`. This PR adds these reports as github artifacts to be used in the codecov CI tool.

### CODE REVIEW

File: .github/workflows/check-runtime.yaml [hunk 1]
```diff
@@ -47,11 +47,13 @@ jobs:
       - name: Build Runtime test suite for Lightning simulator
         if: ${{ matrix.simulator == 'lightning' }}
         run: |
-            cmake -S runtime -B build -DCMAKE_BUILD_TYPE=Release \
+            cmake -S runtime -B build -DCMAKE_BUILD_TYPE=Debug \
```
Comment (dime10) [inline .github/workflows/check-runtime.yaml:50]: Debug is required for coverage checks I assume? :)

**Review by dime10 — COMMENTED** [review]
Would the upload to codecov happen in a separate workflow?

Or should we use the opportunity to move `check-runtime.yaml` into a new job inside `check-catalyst.yaml` and then add a final job to upload to codecov?
Might be easier to manage job dependencies that way.

File: .github/workflows/check-runtime.yaml (hunk 1 above)
Comment (maliasadi) [reply .github/workflows/check-runtime.yaml:50]: Yep, `gcov` required the C++ package to build in the release mode

**Comment by maliasadi** [comment]
> Or should we use the opportunity to move check-runtime.yaml into a new job inside check-catalyst.yaml and then add a final job to upload to codecov?

As we should build the runtime in Debug mode for codecov, we may keep it separate from checking the catalyst. It's good to have it built in both Release and Debug modes. What's your thoughts?

> Would the upload to codecov happen in a separate workflow?

Yes, it's added as separate jobs in these actions.

**Comment by maliasadi** [comment]
> Well it wouldn't change anything regarding how anything is build, I just mean copy-paste the current job in check-runtime.yaml into check-catalyst.yaml. 

Yeah, I've got what you suggested! Reusing the runtime build for the frontend is indeed a good idea if the build options for check-runtime and check-catalist are the same. Here, we need to build the runtime in the Debug mode for C++ code coverage while building it in the Release mode for check-catalyst. Also, we don't have any use case for Kokkos backend from the frontend yet. Building the runtime also takes in less than 5min.  
 
> I see, and that works fine if the reports are uploaded separately, both files don't need to be uploaded together?

It should be fine! I'll double-check.

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+comment:
```
Comment (dime10) [inline codecov.yml:1]: let's rename this file `.codecov.yml` :)

**Review by dime10 — APPROVED** [review]
Perfect, thanks @maliasadi ! 🎉

**Pull request merged**